### PR TITLE
Math Magicians: refactor with hooks

### PR DESCRIPTION
**Pull request by afcastaneda223**
# Documentation

All the documentation is on the README file.

## Updates:

- Refactor the Calculator component (components/Calculator.js) from a class-based component to a functional component with hooks.

## Notes:

Coded by @afcastaneda223

✅  - No linters errors were detected.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,58 +1,60 @@
...
+    if (total === undefined) {
+      total = object.total;
+    }
+    if (next === undefined) {
+      next = object.next;
+    }
+    if (operation === undefined) {
+      operation = object.operation;
+    }
+
```
Comment (henatan99) [inline src/components/Calculator.js:21]: - [ ] When an operator is clicked twice, an error is thrown on the browser. I think you can add a condition here to handle this error. It is required to make sure the app behaves as expected while refactoring. :pray: 
image

**Review by henatan99 — CHANGES_REQUESTED** [review]
Hi @afcastaneda223 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
- [ ] The purpose of this milestone is to practice functional components and React hooks. I recommend you to make the `App` component a functional component too.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by afcastaneda223** [comment]
> Hi @afcastaneda223 ,
> 
> Good job so far!
> There are some issues that you still need to work on to go to the next project but you are almost there!
> 
> ## Required Changes ♻️
> * [ ]  The purpose of this milestone is to practice functional components and React hooks. I recommend you to make the `App` component a functional component too.
> 
> _Check the comments under the review._
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

image


Hi @henatan99 as you can see on the requirements that actually is not a requirement as far as the other request it breaks because of the logic passed on to logic operate.js which actually receives the data from calculate.js which are the two files provided by microverse so the breaking of the code is because the code provided is faulty

**Review by mricanho — APPROVED** [review]
# Status: ✔️ Project Approved ✔️ 

Hello @afcastaneda223  🔥 

You worked hard in all the instructions that the previous Code Reviewers give you👍🏽 

Now you can merge this branch 🚀 

### 💻 👩🏽‍💻  Happy Coding!  👨🏽‍💻 🖥️

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**